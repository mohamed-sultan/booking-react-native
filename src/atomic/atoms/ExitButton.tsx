import React from 'react';
import {TouchableOpacity, StyleSheet} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import  { useThemeColors } from '@store/useTheme';

const ExitButton = ({onPress}: {onPress: () => void}) => {
  const colors = useThemeColors();
  const styles = getStyles(colors);
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <Icon name="close" size={20} color={colors.PrimaryText} />
    </TouchableOpacity>
  );
};
export default React.memo(ExitButton);

const getStyles = (colors: any) => StyleSheet.create({
  container: {
    alignItems: 'center',
    padding: 9,
    backgroundColor: colors.LightBackground,
    width: 34,
    height: 34,
    borderRadius: 10,
    textAlign: 'center',
  },
  text: {
    fontSize: 15,
    fontWeight: '600',
  },
});

