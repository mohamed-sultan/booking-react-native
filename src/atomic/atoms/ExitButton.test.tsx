import React from 'react';
import { render } from '@testing-library/react-native';
import ExitButton from './ExitButton';

// Mock useThemeColors hook
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    PrimaryText: '#000000',
    LightBackground: '#ffffff',
  }),
}));

// Mock react-native-vector-icons/Ionicons
jest.mock('react-native-vector-icons/Ionicons', () => {
  return (props: any) => <mock-icon {...props} />;
});

describe('ExitButton', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<ExitButton onPress={jest.fn()} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
