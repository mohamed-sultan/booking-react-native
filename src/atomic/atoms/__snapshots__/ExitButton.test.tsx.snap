// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExitButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 10,
      "height": 34,
      "opacity": 1,
      "padding": 9,
      "textAlign": "center",
      "width": 34,
    }
  }
>
  <mock-icon
    color="#000000"
    name="close"
    size={20}
  />
</View>
`;
